code/game/objects/items/grenades/clusterbuster.dm

```diff
@@ -36,38 +39,61 @@
 	name = "clusterbang segment"
 	icon = 'icons/obj/grenade.dmi'
 	icon_state = "clusterbang_segment"
+	base_state = "clusterbang_segment"
 
-/obj/item/grenade/clusterbuster/segment/New(var/loc, var/payload_type = /obj/item/grenade/flashbang/cluster)
+/obj/item/grenade/clusterbuster/segment/Initialize(mapload, obj/item/grenade/clusterbuster/base)
 	..()
-	icon_state = "clusterbang_segment_active"
-	payload = payload_type
-	active = 1
+	if(base)
+		name = "[base.name] segment"
+		base_state = "[base.base_state]_segment"
+		icon_state = base_state
+		payload_spawner = base.payload_spawner
+		payload = base.payload
+		prime_sound = base.prime_sound
+		min_spawned = base.min_spawned
+		max_spawned = base.max_spawned
+	icon_state = "[base_state]_active"
+	active = TRUE
 	walk_away(src,loc,rand(1,4))
 	addtimer(CALLBACK(src, .proc/prime), rand(15,60))
 
 /obj/item/grenade/clusterbuster/segment/prime()
-
-	new /obj/effect/payload_spawner(loc, payload, rand(4,8))
-
-	playsound(loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
-
+	new payload_spawner(loc, payload, rand(min_spawned,max_spawned))
+	playsound(loc, prime_sound, 75, 1, -3)
 	qdel(src)
 
 //////////////////////////////////
 //The payload spawner effect
 /////////////////////////////////
-/obj/effect/payload_spawner/New(var/turf/newloc,var/type, var/numspawned as num)
-
-	for(var/loop = numspawned ,loop > 0, loop--)
+/obj/effect/payload_spawner/Initialize(mapload, type, numspawned)
+	for(var/loop in 1 to numspawned)
```

Comment:
..() first

Reply:
. = ..()?

Reply:
no because you're gonna return something else later

Reply:
derp